src/xmipp/libraries/reconstruction_cuda/cuda_volume_restoration_kernels.cu

```diff
@@ -0,0 +1,197 @@
+#include "cuda_cdf.cu"
+
+#include "cuda_basic_math.h"
+
+namespace Gpu {
+
+template< typename T >
+__global__ void computeWeightsKernel(const T* __restrict__ d_Vfiltered1, const T* __restrict__ d_Vfiltered2, T* __restrict__ d_V1r, T* __restrict__ d_V2r,
+ 						T* __restrict__ d_S, const T* __restrict__ d_x, const T* __restrict__ d_probXLessThanx, const T* __restrict__ d_V, size_t volume_size, size_t Nsteps,
+			T weightPower, int weightFun) {
+
+	int n = blockIdx.x * blockDim.x + threadIdx.x;
+
+	if (n >= volume_size) {
+		return;
+	}
+
+	T minVal = d_V[0];
+	T maxVal = d_V[volume_size - 1];
+
+	T f1 = d_Vfiltered1[n];
+	T e1 = f1 * f1;
+	T w1 = Gpu::getCDFProbability(e1, d_x, d_probXLessThanx, Nsteps, minVal, maxVal);
+
+	T f2 = d_Vfiltered2[n];
+	T e2 = f2 * f2;
+	T w2 = Gpu::getCDFProbability(e2, d_x, d_probXLessThanx, Nsteps, minVal, maxVal);
+
+    T weight;
+    switch (weightFun) {
+		case 0: weight = 0.5 * (w1 + w2); break;
+		case 1: weight = min(w1, w2); break;
+		case 2: weight = 0.5 * (w1 + w2) * (1 - abs(w1 - w2)/(w1 + w2)); break;
```

Comment:
missing default case

Reply:
weightFun value is checked before the start of the computation. If it is not 0, 1, 2 an exception is thrown.